tensorflow_model_optimization/python/core/sparsity_tf2/riglpruner_test.py

```diff
@@ -0,0 +1,577 @@
+    self.assertNotAllClose(result2['before'], result2['after'])
+
+
+  @parameterized.parameters(
+     ('random_normal',), ('zeros',)
+  )
+  def testZeroInitGrownConnections(self, reinit_method):
+    # new connections are initialized to zeros
+    weight = tf.Variable(np.linspace(1.0, 100.0, 100))
+    weight_dtype = weight.dtype.base_dtype
+    mask = tf.Variable(
+        tf.ones(weight.get_shape(), dtype=weight_dtype),
+        dtype=weight_dtype)
+    grad = self.grad(weight)
+    sparse_vars = [(mask, weight, grad)]
+    drop_ratio = 0.5
+    sparsity_params = {
+      'pruning_schedule': self.updater(drop_ratio, 0, 4, 1)
+    }
+    sparsity_config = pruning_config.LowMagnitudePruningConfig(**sparsity_params) # dummy
+
+    p = pruner.RiGLPruner(
+      update_schedule=self.updater(drop_ratio, 0, 4, 4), # update iter 1 and 3
+      sparsity=self.target_sparsity,
+      block_size=self.block_size,
+      block_pooling_type=self.block_pooling_type,
+      seed=self.seed,
+      noise_std=self.noise_std,
+      reinit=reinit_method
+    )
+
+    optimizer = pruning_optimizer.PruningOptimizer(
+      tf.keras.optimizers.SGD(learning_rate=0.01), sparsity_config)
+    optimizer.iterations.assign(0)
+
+    p.create_slots(optimizer, weight)
+   
+    def weight_mask_op(pruning_vars):
+      for mask, weight, _ in  pruning_vars:
+        return tf.math.multiply(weight, mask)
+
+    def train(optimizer, weight, sparse_vars):
+      expected1 = tf.TensorArray(tf.bool, size=0, dynamic_size=True, clear_after_read=False)
+      expected2 = tf.TensorArray(tf.bool, size=0, dynamic_size=True, clear_after_read=False)
+      # iterate until all mask updates are complete
+      for i in tf.range(4 + 1):
+        mask_before_update = optimizer.get_slot(weight, 'mask').read_value()
+        weight_before_update = weight.read_value()
+        before_sparsity = tf.math.count_nonzero(mask_before_update, dtype=tf.int32) / tf.size(mask_before_update)
+        grad = p.preprocess_weights(optimizer, weight, self.grad(weight))
+        weight.assign(tf.math.add(weight, sample_noise(i)))
+        grad = grad + sample_noise(i)
+        p.postprocess_weights(optimizer, weight, grad)
+        mask_after_update = optimizer.get_slot(weight, 'mask').read_value()
+        weight_after_update = weight_mask_op(sparse_vars)
+        after_sparsity = tf.math.count_nonzero(mask_after_update, dtype=tf.int32) / tf.size(mask_before_update)
+        # sum the values of the new connections
+        regrown_indices = tf.math.logical_and(mask_before_update == 0, mask_after_update == 1)
+        new_weights1 = weight_after_update[regrown_indices]
+        new_weights2 = weight_after_update[regrown_indices]
+        if reinit_method == 'zeros':
+          expected1 = expected1.write(i,tf.math.reduce_all(new_weights1 == 0))
+        elif reinit_method == 'random_normal':
+          expected2 = expected2.write(i, tf.math.reduce_all(new_weights2 != 0))
```

Comment:
This check is insufficient to verify random_normal is behaving as you expect.You'll need to make the test get the actual stddev too & verify that it's computed from the weight as you expect.